[BUG] test_cast_string_date_invalid_ansi_before_320 failed in databricks runtime

**Describe the bug**
related to https://github.com/NVIDIA/spark-rapids/pull/5459. failed in 312db runtime

```
[2022-05-13T04:29:13.200Z] FAILED ../../src/main/python/cast_test.py::test_cast_string_date_invalid_ansi_before_320
```

```
[2022-05-13T04:29:12.928Z] =================================== FAILURES ===================================
[2022-05-13T04:29:12.928Z] [31m[1m________________ test_cast_string_date_invalid_ansi_before_320 _________________[0m
[2022-05-13T04:29:12.928Z] [gw1] linux -- Python 3.8.13 /databricks/conda/envs/cudf-udf/bin/python
[2022-05-13T04:29:12.928Z] 
[2022-05-13T04:29:12.928Z]     @pytest.mark.skipif(not is_before_spark_320(), reason="ansi cast(string as date) throws exception only in 3.2.0+")
[2022-05-13T04:29:12.928Z]     def test_cast_string_date_invalid_ansi_before_320():
[2022-05-13T04:29:12.928Z]         data_rows = [(v,) for v in values_string_to_data]
[2022-05-13T04:29:12.928Z] >       assert_gpu_and_cpu_are_equal_collect(
[2022-05-13T04:29:12.928Z]             lambda spark: spark.createDataFrame(data_rows, "a string").select(f.col('a').cast(DateType())),
[2022-05-13T04:29:12.928Z]             conf={'spark.rapids.sql.hasExtendedYearValues': 'false',
[2022-05-13T04:29:12.928Z]                   'spark.sql.ansi.enabled': 'true'}, )
[2022-05-13T04:29:12.928Z] 
[2022-05-13T04:29:12.928Z] [1m[31m../../src/main/python/cast_test.py[0m:90: 
[2022-05-13T04:29:12.928Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2022-05-13T04:29:12.928Z] [1m[31m../../src/main/python/asserts.py[0m:508: in assert_gpu_and_cpu_are_equal_collect
[2022-05-13T04:29:12.928Z]     _assert_gpu_and_cpu_are_equal(func, 'COLLECT', conf=conf, is_cpu_first=is_cpu_first)
[2022-05-13T04:29:12.928Z] [1m[31m../../src/main/python/asserts.py[0m:427: in _assert_gpu_and_cpu_are_equal
[2022-05-13T04:29:12.928Z]     run_on_cpu()
[2022-05-13T04:29:12.928Z] [1m[31m../../src/main/python/asserts.py[0m:413: in run_on_cpu
[2022-05-13T04:29:12.928Z]     from_cpu = with_cpu_session(bring_back, conf=conf)
[2022-05-13T04:29:12.929Z] [1m[31m../../src/main/python/spark_session.py[0m:114: in with_cpu_session
[2022-05-13T04:29:12.929Z]     return with_spark_session(func, conf=copy)
[2022-05-13T04:29:12.929Z] [1m[31m../../src/main/python/spark_session.py[0m:98: in with_spark_session
[2022-05-13T04:29:12.929Z]     ret = func(_spark)
[2022-05-13T04:29:12.929Z] [1m[31m../../src/main/python/asserts.py[0m:201: in <lambda>
[2022-05-13T04:29:12.929Z]     bring_back = lambda spark: limit_func(spark).collect()
[2022-05-13T04:29:12.929Z] [1m[31m/databricks/spark/python/pyspark/sql/dataframe.py[0m:697: in collect
[2022-05-13T04:29:12.929Z]     sock_info = self._jdf.collectToPython()
[2022-05-13T04:29:12.929Z] [1m[31m/databricks/spark/python/lib/py4j-0.10.9-src.zip/py4j/java_gateway.py[0m:1304: in __call__
[2022-05-13T04:29:12.929Z]     return_value = get_return_value(
[2022-05-13T04:29:12.929Z] [1m[31m/databricks/spark/python/pyspark/sql/utils.py[0m:117: in deco
[2022-05-13T04:29:12.929Z]     return f(*a, **kw)
[2022-05-13T04:29:12.929Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2022-05-13T04:29:12.929Z] 
[2022-05-13T04:29:12.929Z] answer = 'xro144087'
[2022-05-13T04:29:12.929Z] gateway_client = <py4j.java_gateway.GatewayClient object at 0x7faad0ba0fa0>
[2022-05-13T04:29:12.929Z] target_id = 'o144084', name = 'collectToPython'
[2022-05-13T04:29:12.929Z] 
[2022-05-13T04:29:12.929Z]     def get_return_value(answer, gateway_client, target_id=None, name=None):
[2022-05-13T04:29:12.929Z]         """Converts an answer received from the Java gateway into a Python object.
[2022-05-13T04:29:12.929Z]     
[2022-05-13T04:29:12.929Z]         For example, string representation of integers are converted to Python
[2022-05-13T04:29:12.929Z]         integer, string representation of objects are converted to JavaObject
[2022-05-13T04:29:12.929Z]         instances, etc.
[2022-05-13T04:29:12.929Z]     
[2022-05-13T04:29:12.929Z]         :param answer: the string returned by the Java gateway
[2022-05-13T04:29:12.929Z]         :param gateway_client: the gateway client used to communicate with the Java
[2022-05-13T04:29:12.929Z]             Gateway. Only necessary if the answer is a reference (e.g., object,
[2022-05-13T04:29:12.929Z]             list, map)
[2022-05-13T04:29:12.929Z]         :param target_id: the name of the object from which the answer comes from
[2022-05-13T04:29:12.929Z]             (e.g., *object1* in `object1.hello()`). Optional.
[2022-05-13T04:29:12.929Z]         :param name: the name of the member from which the answer comes from
[2022-05-13T04:29:12.929Z]             (e.g., *hello* in `object1.hello()`). Optional.
[2022-05-13T04:29:12.929Z]         """
[2022-05-13T04:29:12.929Z]         if is_error(answer)[0]:
[2022-05-13T04:29:12.929Z]             if len(answer) > 1:
[2022-05-13T04:29:12.929Z]                 type = answer[1]
[2022-05-13T04:29:12.929Z]                 value = OUTPUT_CONVERTER[type](answer[2:], gateway_client)
[2022-05-13T04:29:12.929Z]                 if answer[1] == REFERENCE_TYPE:
[2022-05-13T04:29:12.929Z] >                   raise Py4JJavaError(
[2022-05-13T04:29:12.929Z]                         "An error occurred while calling {0}{1}{2}.\n".
[2022-05-13T04:29:12.929Z]                         format(target_id, ".", name), value)
[2022-05-13T04:29:12.929Z] [1m[31mE                   py4j.protocol.Py4JJavaError: An error occurred while calling o144084.collectToPython.[0m
[2022-05-13T04:29:12.929Z] [1m[31mE                   : org.apache.spark.SparkException: Job aborted due to stage failure: Task 2 in stage 2498.0 failed 1 times, most recent failure: Lost task 2.0 in stage 2498.0 (TID 7880) (10.2.128.4 executor driver): java.time.DateTimeException: Cannot cast not numbers to DateType.[0m
[2022-05-13T04:29:12.929Z] [1m[31mE                   	at org.apache.spark.sql.catalyst.util.DateTimeUtils$.$anonfun$stringToDateAnsi$1(DateTimeUtils.scala:503)[0m
[2022-05-13T04:29:12.929Z] [1m[31mE                   	at scala.Option.getOrElse(Option.scala:189)[0m
[2022-05-13T04:29:12.929Z] [1m[31mE                   	at org.apache.spark.sql.catalyst.util.DateTimeUtils$.stringToDateAnsi(DateTimeUtils.scala:503)[0m
[2022-05-13T04:29:12.929Z] [1m[31mE                   	at org.apache.spark.sql.catalyst.util.DateTimeUtils.stringToDateAnsi(DateTimeUtils.scala)[0m
[2022-05-13T04:29:12.929Z] [1m[31mE                   	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)[0m
[2022-05-13T04:29:12.929Z] [1m[31mE                   	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)[0m
[2022-05-13T04:29:12.929Z] [1m[31mE                   	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:757)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.UnsafeRowBatchUtils$.encodeUnsafeRows(UnsafeRowBatchUtils.scala:80)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.Collector.$anonfun$processFunc$1(Collector.scala:178)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.scheduler.ResultTask.$anonfun$runTask$3(ResultTask.scala:75)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.scheduler.ResultTask.$anonfun$runTask$1(ResultTask.scala:75)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:55)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:150)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.$anonfun$run$1(Task.scala:119)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.run(Task.scala:91)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$13(Executor.scala:813)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1646)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:816)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:672)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at java.lang.Thread.run(Thread.java:748)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   [0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   Driver stacktrace:[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2828)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2775)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2769)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2769)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1305)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1305)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at scala.Option.foreach(Option.scala:407)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1305)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:3036)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2977)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2965)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:1067)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.SparkContext.runJobInternal(SparkContext.scala:2477)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.Collector.runSparkJobs(Collector.scala:264)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.Collector.collect(Collector.scala:299)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.Collector$.collect(Collector.scala:82)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.Collector$.collect(Collector.scala:88)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.InternalRowFormat$.collect(cachedSparkResults.scala:75)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.InternalRowFormat$.collect(cachedSparkResults.scala:62)[0m
[2022-05-13T04:29:12.930Z] [1m[31mE                   	at org.apache.spark.sql.execution.ResultCacheManager.$anonfun$getOrComputeResultInternal$1(ResultCacheManager.scala:510)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at scala.Option.getOrElse(Option.scala:189)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.execution.ResultCacheManager.getOrComputeResultInternal(ResultCacheManager.scala:509)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.execution.ResultCacheManager.getOrComputeResult(ResultCacheManager.scala:397)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.execution.ResultCacheManager.getOrComputeResult(ResultCacheManager.scala:372)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.execution.SparkPlan.executeCollectResult(SparkPlan.scala:406)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.$anonfun$collectToPython$1(Dataset.scala:3613)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:3825)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$5(SQLExecution.scala:130)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:273)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$1(SQLExecution.scala:104)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:854)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.withCustomExecutionEnv(SQLExecution.scala:77)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:223)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3823)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.collectToPython(Dataset.scala:3611)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at sun.reflect.GeneratedMethodAccessor138.invoke(Unknown Source)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at java.lang.reflect.Method.invoke(Method.java:498)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:380)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at py4j.Gateway.invoke(Gateway.java:295)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at py4j.commands.CallCommand.execute(CallCommand.java:79)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at py4j.GatewayConnection.run(GatewayConnection.java:251)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at java.lang.Thread.run(Thread.java:748)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   Caused by: java.time.DateTimeException: Cannot cast not numbers to DateType.[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.catalyst.util.DateTimeUtils$.$anonfun$stringToDateAnsi$1(DateTimeUtils.scala:503)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at scala.Option.getOrElse(Option.scala:189)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.catalyst.util.DateTimeUtils$.stringToDateAnsi(DateTimeUtils.scala:503)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.catalyst.util.DateTimeUtils.stringToDateAnsi(DateTimeUtils.scala)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:757)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.UnsafeRowBatchUtils$.encodeUnsafeRows(UnsafeRowBatchUtils.scala:80)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.Collector.$anonfun$processFunc$1(Collector.scala:178)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.scheduler.ResultTask.$anonfun$runTask$3(ResultTask.scala:75)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.scheduler.ResultTask.$anonfun$runTask$1(ResultTask.scala:75)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:55)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:150)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.$anonfun$run$1(Task.scala:119)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.run(Task.scala:91)[0m
[2022-05-13T04:29:12.931Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$13(Executor.scala:813)[0m
[2022-05-13T04:29:12.932Z] [1m[31mE                   	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1646)[0m
[2022-05-13T04:29:12.932Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:816)[0m
[2022-05-13T04:29:12.932Z] [1m[31mE                   	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)[0m
[2022-05-13T04:29:12.932Z] [1m[31mE                   	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)[0m
[2022-05-13T04:29:12.932Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:672)[0m
[2022-05-13T04:29:12.932Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)[0m
[2022-05-13T04:29:12.932Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)[0m
[2022-05-13T04:29:12.932Z] [1m[31mE                   	... 1 more[0m
[2022-05-13T04:29:12.932Z] 
[2022-05-13T04:29:12.932Z] [1m[31m/databricks/spark/python/lib/py4j-0.10.9-src.zip/py4j/protocol.py[0m:326: Py4JJavaError
[2022-05-13T04:29:12.932Z] ----------------------------- Captured stdout call -----------------------------
[2022-05-13T04:29:12.932Z] ### CPU RUN ###
[2022-05-13T04:29:12.932Z] ----------------------------- Captured stderr call -----------------------------
[2022-05-13T04:29:12.932Z] 22/05/13 03:27:30 ERROR Executor: Exception in task 1.0 in stage 2498.0 (TID 7879)
[2022-05-13T04:29:12.932Z] java.time.DateTimeException: Cannot cast 1970-01 A to DateType.
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.sql.catalyst.util.DateTimeUtils$.$anonfun$stringToDateAnsi$1(DateTimeUtils.scala:503)
[2022-05-13T04:29:12.932Z] 	at scala.Option.getOrElse(Option.scala:189)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.sql.catalyst.util.DateTimeUtils$.stringToDateAnsi(DateTimeUtils.scala:503)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.sql.catalyst.util.DateTimeUtils.stringToDateAnsi(DateTimeUtils.scala)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:757)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.sql.execution.collect.UnsafeRowBatchUtils$.encodeUnsafeRows(UnsafeRowBatchUtils.scala:80)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.sql.execution.collect.Collector.$anonfun$processFunc$1(Collector.scala:178)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.scheduler.ResultTask.$anonfun$runTask$3(ResultTask.scala:75)
[2022-05-13T04:29:12.932Z] 	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.scheduler.ResultTask.$anonfun$runTask$1(ResultTask.scala:75)
[2022-05-13T04:29:12.932Z] 	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:55)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:150)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.scheduler.Task.$anonfun$run$1(Task.scala:119)
[2022-05-13T04:29:12.932Z] 	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.scheduler.Task.run(Task.scala:91)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$13(Executor.scala:813)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1646)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:816)
[2022-05-13T04:29:12.932Z] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2022-05-13T04:29:12.932Z] 	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:672)
[2022-05-13T04:29:12.932Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2022-05-13T04:29:12.932Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2022-05-13T04:29:12.932Z] 	at java.lang.Thread.run(Thread.java:748)
[2022-05-13T04:29:12.932Z] 22/05/13 03:27:30 ERROR Executor: Exception in task 2.0 in stage 2498.0 (TID 7880)
[2022-05-13T04:29:12.932Z] java.time.DateTimeException: Cannot cast not numbers to DateType.
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.sql.catalyst.util.DateTimeUtils$.$anonfun$stringToDateAnsi$1(DateTimeUtils.scala:503)
[2022-05-13T04:29:12.932Z] 	at scala.Option.getOrElse(Option.scala:189)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.sql.catalyst.util.DateTimeUtils$.stringToDateAnsi(DateTimeUtils.scala:503)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.sql.catalyst.util.DateTimeUtils.stringToDateAnsi(DateTimeUtils.scala)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:757)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.sql.execution.collect.UnsafeRowBatchUtils$.encodeUnsafeRows(UnsafeRowBatchUtils.scala:80)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.sql.execution.collect.Collector.$anonfun$processFunc$1(Collector.scala:178)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.scheduler.ResultTask.$anonfun$runTask$3(ResultTask.scala:75)
[2022-05-13T04:29:12.932Z] 	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
[2022-05-13T04:29:12.932Z] 	at org.apache.spark.scheduler.ResultTask.$anonfun$runTask$1(ResultTask.scala:75)
[2022-05-13T04:29:12.933Z] 	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:55)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:150)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.scheduler.Task.$anonfun$run$1(Task.scala:119)
[2022-05-13T04:29:12.933Z] 	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.scheduler.Task.run(Task.scala:91)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$13(Executor.scala:813)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1646)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:816)
[2022-05-13T04:29:12.933Z] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2022-05-13T04:29:12.933Z] 	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:672)
[2022-05-13T04:29:12.933Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2022-05-13T04:29:12.933Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2022-05-13T04:29:12.933Z] 	at java.lang.Thread.run(Thread.java:748)
[2022-05-13T04:29:12.933Z] 22/05/13 03:27:30 WARN TaskSetManager: Lost task 2.0 in stage 2498.0 (TID 7880) (10.2.128.4 executor driver): java.time.DateTimeException: Cannot cast not numbers to DateType.
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.sql.catalyst.util.DateTimeUtils$.$anonfun$stringToDateAnsi$1(DateTimeUtils.scala:503)
[2022-05-13T04:29:12.933Z] 	at scala.Option.getOrElse(Option.scala:189)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.sql.catalyst.util.DateTimeUtils$.stringToDateAnsi(DateTimeUtils.scala:503)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.sql.catalyst.util.DateTimeUtils.stringToDateAnsi(DateTimeUtils.scala)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:757)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.sql.execution.collect.UnsafeRowBatchUtils$.encodeUnsafeRows(UnsafeRowBatchUtils.scala:80)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.sql.execution.collect.Collector.$anonfun$processFunc$1(Collector.scala:178)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.scheduler.ResultTask.$anonfun$runTask$3(ResultTask.scala:75)
[2022-05-13T04:29:12.933Z] 	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.scheduler.ResultTask.$anonfun$runTask$1(ResultTask.scala:75)
[2022-05-13T04:29:12.933Z] 	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:55)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:150)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.scheduler.Task.$anonfun$run$1(Task.scala:119)
[2022-05-13T04:29:12.933Z] 	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.scheduler.Task.run(Task.scala:91)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$13(Executor.scala:813)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1646)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:816)
[2022-05-13T04:29:12.933Z] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2022-05-13T04:29:12.933Z] 	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:672)
[2022-05-13T04:29:12.933Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2022-05-13T04:29:12.933Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2022-05-13T04:29:12.933Z] 	at java.lang.Thread.run(Thread.java:748)
[2022-05-13T04:29:12.933Z] 
[2022-05-13T04:29:12.933Z] 22/05/13 03:27:30 ERROR TaskSetManager: Task 2 in stage 2498.0 failed 1 times; aborting job
[2022-05-13T04:29:12.933Z] 22/05/13 03:27:30 WARN TaskSetManager: Lost task 1.0 in stage 2498.0 (TID 7879) (10.2.128.4 executor driver): java.time.DateTimeException: Cannot cast 1970-01 A to DateType.
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.sql.catalyst.util.DateTimeUtils$.$anonfun$stringToDateAnsi$1(DateTimeUtils.scala:503)
[2022-05-13T04:29:12.933Z] 	at scala.Option.getOrElse(Option.scala:189)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.sql.catalyst.util.DateTimeUtils$.stringToDateAnsi(DateTimeUtils.scala:503)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.sql.catalyst.util.DateTimeUtils.stringToDateAnsi(DateTimeUtils.scala)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2022-05-13T04:29:12.933Z] 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:757)
[2022-05-13T04:29:12.934Z] 	at org.apache.spark.sql.execution.collect.UnsafeRowBatchUtils$.encodeUnsafeRows(UnsafeRowBatchUtils.scala:80)
[2022-05-13T04:29:12.934Z] 	at org.apache.spark.sql.execution.collect.Collector.$anonfun$processFunc$1(Collector.scala:178)
[2022-05-13T04:29:12.934Z] 	at org.apache.spark.scheduler.ResultTask.$anonfun$runTask$3(ResultTask.scala:75)
[2022-05-13T04:29:12.934Z] 	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
[2022-05-13T04:29:12.934Z] 	at org.apache.spark.scheduler.ResultTask.$anonfun$runTask$1(ResultTask.scala:75)
[2022-05-13T04:29:12.934Z] 	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
[2022-05-13T04:29:12.934Z] 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:55)
[2022-05-13T04:29:12.934Z] 	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:150)
[2022-05-13T04:29:12.934Z] 	at org.apache.spark.scheduler.Task.$anonfun$run$1(Task.scala:119)
[2022-05-13T04:29:12.934Z] 	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
[2022-05-13T04:29:12.934Z] 	at org.apache.spark.scheduler.Task.run(Task.scala:91)
[2022-05-13T04:29:12.934Z] 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$13(Executor.scala:813)
[2022-05-13T04:29:12.934Z] 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1646)
[2022-05-13T04:29:12.934Z] 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:816)
[2022-05-13T04:29:12.934Z] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2022-05-13T04:29:12.934Z] 	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
[2022-05-13T04:29:12.934Z] 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:672)
[2022-05-13T04:29:12.934Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2022-05-13T04:29:12.934Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2022-05-13T04:29:12.934Z] 	at java.lang.Thread.run(Thread.java:748)
```

--- a/integration_tests/src/main/python/cast_test.py
+++ b/integration_tests/src/main/python/cast_test.py
@@ -16,7 +16,7 @@ import pytest
 
 from asserts import assert_gpu_and_cpu_are_equal_collect, assert_gpu_and_cpu_are_equal_sql, assert_gpu_and_cpu_error, assert_gpu_fallback_collect, assert_py4j_exception
 from data_gen import *
-from spark_session import is_before_spark_320, is_before_spark_330, with_gpu_session
+from spark_session import is_before_spark_320, is_before_spark_330, is_databricks91_or_later, with_gpu_session
 from marks import allow_non_gpu, approximate_float
 from pyspark.sql.types import *
 from spark_init_internal import spark_version
@@ -83,14 +83,35 @@ valid_values_string_to_date = ['2001', ' 2001 ', '1970-01', ' 1970-1 ',
                                ]
 values_string_to_data = invalid_values_string_to_date + valid_values_string_to_date
 
-# test Spark Spark versions < 3.2.0, ANSI mode
-@pytest.mark.skipif(not is_before_spark_320(), reason="ansi cast(string as date) throws exception only in 3.2.0+")
+# Spark 320+ and db312+ support Ansi mode when casting string to date
+# This means an exception will be thrown when casting invalid string to date on Spark 320+ or db312+
+# test Spark versions < 3.2.0 and non db, ANSI mode
+@pytest.mark.skipif((not is_before_spark_320()) or is_databricks91_or_later(), reason="ansi cast(string as date) throws exception only in 3.2.0+ or db")
 def test_cast_string_date_invalid_ansi_before_320():
     data_rows = [(v,) for v in values_string_to_data]
     assert_gpu_and_cpu_are_equal_collect(
         lambda spark: spark.createDataFrame(data_rows, "a string").select(f.col('a').cast(DateType())),
         conf={'spark.rapids.sql.hasExtendedYearValues': 'false',
               'spark.sql.ansi.enabled': 'true'}, )
+
+# test db312+, ANSI mode, all db versions supports Ansi mode when casting string to date
+@pytest.mark.skipif(not is_databricks91_or_later(), reason="Spark versions(< 320) not support Ansi mode when casting string to date")
+@pytest.mark.parametrize('invalid', invalid_values_string_to_date)
+def test_cast_string_date_invalid_ansi_dbs(invalid):
+    assert_gpu_and_cpu_error(
+        lambda spark: spark.createDataFrame([(invalid,)], "a string").select(f.col('a').cast(DateType())).collect(),
+        conf={'spark.rapids.sql.hasExtendedYearValues': 'false',
+              'spark.sql.ansi.enabled': 'true'},
+        error_message="DateTimeException")
+
+# test db312+, ANSI mode, valid values
+@pytest.mark.skipif(not is_databricks91_or_later(), reason="Spark versions(< 320) not support Ansi mode when casting string to date")
+def test_cast_string_date_valid_ansi_dbs():
+    data_rows = [(v,) for v in valid_values_string_to_date]
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: spark.createDataFrame(data_rows, "a string").select(f.col('a').cast(DateType())),
+        conf={'spark.rapids.sql.hasExtendedYearValues': 'false',
+              'spark.sql.ansi.enabled': 'true'})
 
 # test Spark versions >= 320, ANSI mode
 @pytest.mark.skipif(is_before_spark_320(), reason="ansi cast(string as date) throws exception only in 3.2.0+")

--- a/integration_tests/src/main/python/cast_test.py
+++ b/integration_tests/src/main/python/cast_test.py
@@ -97,7 +97,7 @@ def test_cast_string_date_invalid_ansi_before_320():
 # test databricks, ANSI mode, all databricks versions supports Ansi mode when casting string to date
 @pytest.mark.skipif(not is_databricks91_or_later(), reason="Spark versions(< 320) not support Ansi mode when casting string to date")
 @pytest.mark.parametrize('invalid', invalid_values_string_to_date)
-def test_cast_string_date_invalid_ansi_dbs(invalid):
+def test_cast_string_date_invalid_ansi_databricks(invalid):
     assert_gpu_and_cpu_error(
         lambda spark: spark.createDataFrame([(invalid,)], "a string").select(f.col('a').cast(DateType())).collect(),
         conf={'spark.rapids.sql.hasExtendedYearValues': 'false',
@@ -106,7 +106,7 @@ def test_cast_string_date_invalid_ansi_dbs(invalid):
 
 # test databricks, ANSI mode, valid values
 @pytest.mark.skipif(not is_databricks91_or_later(), reason="Spark versions(< 320) not support Ansi mode when casting string to date")
-def test_cast_string_date_valid_ansi_dbs():
+def test_cast_string_date_valid_ansi_databricks():
     data_rows = [(v,) for v in valid_values_string_to_date]
     assert_gpu_and_cpu_are_equal_collect(
         lambda spark: spark.createDataFrame(data_rows, "a string").select(f.col('a').cast(DateType())),

--- a/integration_tests/src/main/python/cast_test.py
+++ b/integration_tests/src/main/python/cast_test.py
@@ -83,9 +83,9 @@ valid_values_string_to_date = ['2001', ' 2001 ', '1970-01', ' 1970-1 ',
                                ]
 values_string_to_data = invalid_values_string_to_date + valid_values_string_to_date
 
-# Spark 320+ and db312+ support Ansi mode when casting string to date
-# This means an exception will be thrown when casting invalid string to date on Spark 320+ or db312+
-# test Spark versions < 3.2.0 and non db, ANSI mode
+# Spark 320+ and databricks support Ansi mode when casting string to date
+# This means an exception will be thrown when casting invalid string to date on Spark 320+ or databricks
+# test Spark versions < 3.2.0 and non databricks, ANSI mode
 @pytest.mark.skipif((not is_before_spark_320()) or is_databricks91_or_later(), reason="ansi cast(string as date) throws exception only in 3.2.0+ or db")
 def test_cast_string_date_invalid_ansi_before_320():
     data_rows = [(v,) for v in values_string_to_data]
@@ -94,7 +94,7 @@ def test_cast_string_date_invalid_ansi_before_320():
         conf={'spark.rapids.sql.hasExtendedYearValues': 'false',
               'spark.sql.ansi.enabled': 'true'}, )
 
-# test db312+, ANSI mode, all db versions supports Ansi mode when casting string to date
+# test databricks, ANSI mode, all databricks versions supports Ansi mode when casting string to date
 @pytest.mark.skipif(not is_databricks91_or_later(), reason="Spark versions(< 320) not support Ansi mode when casting string to date")
 @pytest.mark.parametrize('invalid', invalid_values_string_to_date)
 def test_cast_string_date_invalid_ansi_dbs(invalid):
@@ -104,7 +104,7 @@ def test_cast_string_date_invalid_ansi_dbs(invalid):
               'spark.sql.ansi.enabled': 'true'},
         error_message="DateTimeException")
 
-# test db312+, ANSI mode, valid values
+# test databricks, ANSI mode, valid values
 @pytest.mark.skipif(not is_databricks91_or_later(), reason="Spark versions(< 320) not support Ansi mode when casting string to date")
 def test_cast_string_date_valid_ansi_dbs():
     data_rows = [(v,) for v in valid_values_string_to_date]

--- a/integration_tests/src/main/python/parquet_test.py
+++ b/integration_tests/src/main/python/parquet_test.py
@@ -878,14 +878,14 @@ def test_parquet_check_schema_compatibility(spark_tmp_path):
     assert_gpu_and_cpu_error(
         lambda spark: spark.read.schema(read_int_as_long).parquet(data_path).collect(),
         conf={},
-        error_message='Parquet column cannot be converted in')
+        error_message='Parquet column cannot be converted')
 
     read_dec32_as_dec64 = StructType(
         [StructField('int', IntegerType()), StructField('dec32', DecimalType(15, 10))])
     assert_gpu_and_cpu_error(
         lambda spark: spark.read.schema(read_dec32_as_dec64).parquet(data_path).collect(),
         conf={},
-        error_message='Parquet column cannot be converted in')
+        error_message='Parquet column cannot be converted')
 
 
 # For nested types, GPU throws incompatible exception with a different message from CPU.
@@ -902,32 +902,32 @@ def test_parquet_check_schema_compatibility_nested_types(spark_tmp_path):
     assert_py4j_exception(
         lambda: with_gpu_session(
             lambda spark: spark.read.schema(read_array_long_as_int).parquet(data_path).collect()),
-        error_message='Parquet column cannot be converted in')
+        error_message='Parquet column cannot be converted')
 
     read_arr_arr_int_as_long = StructType(
         [StructField('array_array_int', ArrayType(ArrayType(LongType())))])
     assert_py4j_exception(
         lambda: with_gpu_session(
             lambda spark: spark.read.schema(read_arr_arr_int_as_long).parquet(data_path).collect()),
-        error_message='Parquet column cannot be converted in')
+        error_message='Parquet column cannot be converted')
 
     read_struct_flt_as_dbl = StructType([StructField(
         'struct_float', StructType([StructField('f', DoubleType())]))])
     assert_py4j_exception(
         lambda: with_gpu_session(
             lambda spark: spark.read.schema(read_struct_flt_as_dbl).parquet(data_path).collect()),
-        error_message='Parquet column cannot be converted in')
+        error_message='Parquet column cannot be converted')
 
     read_struct_arr_int_as_long = StructType([StructField(
         'struct_array_int', StructType([StructField('a', ArrayType(LongType()))]))])
     assert_py4j_exception(
         lambda: with_gpu_session(
             lambda spark: spark.read.schema(read_struct_arr_int_as_long).parquet(data_path).collect()),
-        error_message='Parquet column cannot be converted in')
+        error_message='Parquet column cannot be converted')
 
     read_map_str_str_as_str_int = StructType([StructField(
         'map', MapType(StringType(), IntegerType()))])
     assert_py4j_exception(
         lambda: with_gpu_session(
             lambda spark: spark.read.schema(read_map_str_str_as_str_int).parquet(data_path).collect()),
-        error_message='Parquet column cannot be converted in')
+        error_message='Parquet column cannot be converted')

--- a/sql-plugin/src/main/31xdb/scala/com/nvidia/spark/rapids/shims/Spark31XdbShims.scala
+++ b/sql-plugin/src/main/31xdb/scala/com/nvidia/spark/rapids/shims/Spark31XdbShims.scala
@@ -59,9 +59,12 @@ abstract class Spark31XdbShims extends Spark31XdbShimsBase with Logging {
     GpuOverrides.expr[Cast](
         "Convert a column of one type of data into another type",
         new CastChecks(),
+        // 312db supports Ansi mode when casting string to date, this means that an exception
+        // will be thrown when casting an invalid value to date in Ansi mode.
+        // Set `stringToAnsiDate` = true
         (cast, conf, p, r) => new CastExprMeta[Cast](cast,
           SparkSession.active.sessionState.conf.ansiEnabled, conf, p, r,
-          doFloatToIntCheck = true, stringToAnsiDate = false)),
+          doFloatToIntCheck = true, stringToAnsiDate = true)),
     GpuOverrides.expr[AnsiCast](
       "Convert a column of one type of data into another type",
       new CastChecks {
@@ -112,8 +115,11 @@ abstract class Spark31XdbShims extends Spark31XdbShimsBase with Logging {
         override val udtChecks: TypeSig = none
         override val sparkUdtSig: TypeSig = UDT
       },
+      // 312db supports Ansi mode when casting string to date, this means that an exception
+      // will be thrown when casting an invalid value to date in Ansi mode.
+      // Set `stringToAnsiDate` = true
       (cast, conf, p, r) => new CastExprMeta[AnsiCast](cast, ansiEnabled = true, conf = conf,
-        parent = p, rule = r, doFloatToIntCheck = true, stringToAnsiDate = false)),
+        parent = p, rule = r, doFloatToIntCheck = true, stringToAnsiDate = true)),
     GpuOverrides.expr[Average](
       "Average aggregate operator",
       ExprChecks.fullAgg(

